docs/hack_loki_operator.md

```diff
@@ -0,0 +1,222 @@
+
+### Requirements
+
+* Install [kubectl](https://kubernetes.io/docs/tasks/tools/#kubectl) or [Openshift CLI](https://docs.openshift.com/container-platform/latest/cli_reference/openshift_cli/getting-started-cli.html) for communicating with the cluster. The guide below will be using `kubectl` for the same.
+* Create a running Kubernetes cluster using kind.
+* A container registry that you and your Kubernetes cluster can reach. We recommend  [quay.io](https://quay.io/signin/).
+
+### Installation of Loki Operator
+
+* Build and push the container image and then deploy the operator with:
+
+  ```console
+  make oci-build oci-push deploy REGISTRY_ORG=$YOUR_QUAY_ORG VERSION=latest
+  ```
+
+  where `$YOUR_QUAY_ORG` is your personal [quay.io](http://quay.io/) account where you can push container images.
+
+  The above command will deploy the operator to your active Kubernetes cluster defined by your local [kubeconfig](https://kubernetes.io/docs/concepts/configuration/organize-cluster-access-kubeconfig/). The operator will be running in the `default` namespace.
+
+* You can confirm that the operator is up and running using:
+
+  ```console
+  kubectl get pods
+  ```
+
+* Now create a LokiStack instance to get the various components of Loki up and running:
+
+  ```console
+  kubectl apply -f hack/lokistack_gateway_dev.yaml
+  ```
+
+  This will create `distributor`, `compactor`, `ingester`, `querier`, `query-frontend` and `lokistack-gateway` components.
+
+  Confirm that all are up and running for `deployments` using:
+
+  ```console  
+  kubectl rollout status deployment/<DEPLOYMENT_NAME>
+  ```  
+
+  where `<DEPLOYMENT_NAME>` is the name of the deployment and can be found using:
+
+  ```console
+  kubectl get deployments
+  ```
+
+  Confirm that all are up and running for `statefulsets` using:
+
+  ```console  
+  kubectl rollout status statefulset/<STATEFULSET_NAME>  
+  ```  
+
+  where `<STATEFULSET_NAME>` is the name of the statefulset and can be found using:
+
+  ```console
+  kubectl get statefulsets
+  ```
+
+  _Note:_  `lokistack-gateway` is an optional component deployed as part of Loki Operator. It provides secure access to Loki's distributor (i.e. for pushing logs) and query-frontend (i.e. for querying logs) via consulting an OAuth/OIDC endpoint for the request subject.
+
+  If you don't want `lokistack-gateway` component then you can skip it by removing the `--with-lokistack-gateway` args from the `controller-manager` deployment:
+
+  ```console
+  kubectl edit deployment/controller-manager
+  ```
+
+  Delete the flag `--with-lokistack-gateway` from the `args` section and save the file. This will update the deployment and now you can create LokiStack instance using:
+
+  ```console
+  kubectl apply -f hack/lokistack_dev.yaml
```

Comment:
I see in lokistack_dev.yaml reference to https://github.com/ViaQ/loki-operator/blob/master/hack/lokistack_dev.yaml#L11 **gp2** storage class. This doesn't work with kind

Reply:
Think this is now resolved due to the changes brought in by #101.